etc/spack/defaults/modules.yaml

```diff
@@ -14,26 +14,35 @@
 #   ~/.spack/modules.yaml
 # -------------------------------------------------------------------------
 modules:
-  enable:
-    - tcl
-  prefix_inspections:
-    bin:
-      - PATH
-    man:
-      - MANPATH
-    share/man:
-      - MANPATH
-    share/aclocal:
-      - ACLOCAL_PATH
-    lib/pkgconfig:
-      - PKG_CONFIG_PATH
-    lib64/pkgconfig:
-      - PKG_CONFIG_PATH
-    share/pkgconfig:
-      - PKG_CONFIG_PATH
-    '':
-      - CMAKE_PREFIX_PATH
+  # These are configurations for the module set named "default"
+  default:
```

Comment:
FWIW, I think I might've gotten used to `default` since I wrote this.

Reply:
This is definitely an improvement over the current state, so :heavy_check_mark: from my side. 

I was wondering though if we considered isolating the configuration even further by allowing a list of entries, each of which generates just a single type of modules:
```yaml
modules:
- name: 'tcl-default'
  type: tcl
  root: $spack/share/spack/modules
  [...]
- name: 'lmod-default'
  type: lmod
  root: $spack/share/spack/lmod
  [...]
```
This will simplify handling roots and remove the necessity to use `enable`.

Reply:
I guess a drawback of my suggestion is that having different defaults based on platforms would require more extensive changes to the current code :thinking: Anyhow, left the idea here is that is considered useful.